pkg/istio-agent/agent.go

```diff
@@ -536,8 +536,7 @@ func (a *Agent) startFileWatcher(ctx context.Context, filePath string, handler f
 }
 
 func (a *Agent) initLocalDNSServer() (err error) {
-	// we don't need dns server on gateways
-	if a.cfg.DNSCapture && a.cfg.ProxyType == model.SidecarProxy {
+	if a.cfg.DNSCapture {
```

Comment:
we need a separate flag to explicitly allow it on gateway

Reply:
BY this most users make use of dns proxy may run dns proxy in gateway unexpectedly